src/components/company/CompanyAddForm.tsx

```diff
@@ -0,0 +1,310 @@
+        setForm(oldForm => ({ ...oldForm, [fieldName]: fieldValue }));
+        setFormErrorState(prevErrorState => ({
+            ...prevErrorState,
+            [fieldName]: hasFormFieldError({
+                fieldName,
+                fieldValue,
+                isRequired: requiredFields.indexOf(fieldName) > -1
+            })
+        }));
+
+        if (fieldName === 'name') {
+            const companyId = RegExUtil.conformToId(fieldValue);
+            setForm(oldForm => ({
+                ...oldForm,
+                companyId,
+                description: fieldValue
+            }));
+            setFormErrorState(prevErrorState => ({
+                ...prevErrorState,
+                companyId: hasFormFieldError({
+                    fieldName: 'companyId',
+                    fieldValue: companyId,
+                    isRequired: requiredFields.indexOf('companyId') > -1
+                }),
+                description: hasFormFieldError({
+                    fieldName: 'description',
+                    fieldValue,
+                    isRequired: requiredFields.indexOf('description') > -1
+                })
+            }));
+        }
+    };
+
+    const onBlockMessagingChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const isMessagingBlocked = !e.target.checked;
+        const modifiedLmsSettings = isMessagingBlocked
+            ? {
+                  blockMessaging: true
+              }
+            : { ...DEFAULT_LMS_SETTINGS };
+        setForm(oldForm => ({
+            ...oldForm,
+            settings: {
+                ...oldForm.settings,
+                lmsSettings: modifiedLmsSettings
+            }
+        }));
+    };
+
+    const onCreateClick = () => {
+        const { errorState: modifiedErrorState, hasFormError } =
+            getFormErrorData({
+                form,
+                requiredFields
+            });
+        setFormErrorState(prevErrorState => ({
+            ...prevErrorState,
+            ...modifiedErrorState
+        }));
+
+        if (hasFormError) {
+            return;
+        }
+
+        // TODO: Integrate API (done in next PR)
+    };
+
+    return (
+        <Box
+            px={8}
+            py={6}
+            borderWidth={{ base: 0, sm: 1 }}
+            borderRadius="lg"
+            width="3xl"
+        >
+            <SimpleGrid
+                columns={{ base: 1, sm: 2 }}
+                spacingX={6}
+                spacingY={4}
+                width="100%"
+                alignItems="start"
+            >
+                <GridItem colSpan={{ base: 1, sm: 2 }} mb={4}>
+                    <Text
+                        fontSize="xl"
+                        lineHeight={1.4}
+                        width="100%"
+                        fontWeight={600}
+                    >
+                        Create Company
+                    </Text>
+                </GridItem>
+                {companyFields.map(field => (
```

Comment:
don't follow this way of creating forms, this creates unnecessary complexity. BTDT. This is not scalable.